android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -255,6 +262,57 @@ class BrowserIcons constructor(
         }
     }
 
+    /**
+     * Loads an icon using [BrowserIcons] into the given Composable [content]. Synchronous loading
+     * via an in-memory cache is attempted first, followed by an asynchronous load as a fallback.
+     *
+     * @param url The URL of the website an icon should be loaded for.
+     * @param iconSize The preferred size of the icon that should be loaded.
+     * @param isPrivate Whether this request for this icon came from a private session.
+     * @param content The Composable content block to render the icon.
+     */
+    @Composable
+    fun LoadableImage(
```

Comment:
does this need to still live inside the `BrowserIcons` class? it'll make it more encapsulated as its own thing but also clean up the usage inside of `Favicon`

Reply:
@MozillaNoah my reasoning to keep it in this inside the `BrowserIcons` class is that the fun is using a number of functions provided in the class:

- `private fun desiredSizeForRequest`
- `private fun loadIconMemoryOnly`
- `internal fun loadIconInternalAsync`

By keeping this in the same file it means fewer changes to the existing code which is my preference, wdyt?

Reply:
ahh gotcha gotcha. I totally missed that several class functions were being used here. I think in an ideal world we'd have the Composable in its own file, decoupled from some of the specific nuances of `BrowserIcons`, but that is definitely way out of scope here, having given a full pass on that class.

This is good to go 👍